lib/id.js

```diff
@@ -73,13 +77,46 @@ const generateSID = config => {
 //   config - <Object>, { secret }
 //   sid - <string>, session id
 // Returns: <boolean>
+// Deprecated: this method will be removed in the next major versions.
+//     Use `validateToken()` instead.
 const validateSID = (config, sid) => {
   if (!sid) return false;
   const crc = sid.substr(sid.length - 4);
   const key = sid.substr(0, sid.length - 4);
   return crcSID(config, key) === crc;
 };
 
+// Calculate Token crc
+//   secret - <string>
+//   key - <string>
+// Returns: <string>, crc
+const crcToken = (secret, key) => crypto.createHash('md5')
+  .update(key + secret)
+  .digest('hex')
+  .substring(0, 4);
+
+// Generate random Token
+//   secret <string>
+//   characters <string>
+//   length <number>
+// Returns: <string>, token
+const generateToken = (secret, characters, length) => {
+  const key = generateKey(length - 4, characters);
+  return key + crcToken(secret, key);
+};
+
+// Validate Token
+//   secret - <string>
+//   token - <string>
+// Returns: <boolean>
+const validateToken = (secret, token) => {
+  if (!token) return false;
+  const len = token.length;
+  const crc = token.substr(len - 4);
```

Comment:
`String.prototype.substr()` is considered legacy function, according to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr), I think we can replace it with `String.prototype.slice()` here.